MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs

```diff
@@ -123,12 +120,51 @@ private enum Codec
 
         public static Version MaxVersion { get; } = null;
 
-        public bool ValidateVersion()
+        /// <summary>
+        /// Returns a value indicating which version of the encoder more suitable.
+        /// </summary>
+        /// <param name="source">Source encoding version.</param>
+        /// <param name="destination">Destination encoding version.</param>
+        /// <returns><c>false</c> if source is better, <c>true</c> if destination is better.</returns>
+        public static bool BestVersion(Version source, Version destination)
+        {
+            if (source == null || destination == null)
+            {
+                return source == null;
+            }
+
+            // If either is recommended, then select that.
+            bool sourceRecommended = source >= MinVersion && (MaxVersion == null || source <= MaxVersion);
+            bool destRecommended = destination >= MinVersion && (MaxVersion == null || destination <= MaxVersion);
+
+            if (sourceRecommended)
+            {
+                if (destRecommended)
+                {
+                    // if both are recommended, select the latest version.
+                    return !(source >= destination);
```

Comment:
Why the negation? Wouldn't `source < destination` suffice?

Reply:
the way my brain works! Changed.